### PR TITLE
Custom stub naming

### PR DESCRIPTION
**Pull request by marcingrzejszczak**
Crucial changes:
- `microservice.json` conains the new "stubs" section
- Stub-runner no longer fetches dependencies from `collaborators` endpoint but from `microservice.json`
    - the problem is that the endpoint is customized
    - in spring cloud approach it won't be a json

CC @dkublik , @nurkiewicz , @S2lomon , @cameleeck

### CODE REVIEW

File: micro-deps/micro-deps/src/main/java/com/ofg/infrastructure/discovery/ServicePath.java [hunk 1]
```diff
@@ -19,9 +20,31 @@ public String getPath() {
...
+            return StringUtils.EMPTY;
+        }
         return Iterables.getLast(Arrays.asList(path.split("/")));
     }
 
+    public String getPathToLastName() {
+        if (path == null) {
+            return StringUtils.EMPTY;
+        }
+        int lastOccurenceOfSlash = path.lastIndexOf("/");
```
Comment (cameleeck) [inline micro-deps/micro-deps/src/main/java/com/ofg/infrastructure/discovery/ServicePath.java]: StringUtils.removeEnd() ?

**Comment by mariuszs** [comment]
+1

**Comment by marcingrzejszczak** [comment]
Merged in https://github.com/4finance/micro-infra-spring/commit/fd76ff56552991197654b7cfc56569adc7cd79ed

File: micro-deps/micro-deps/src/main/java/com/ofg/infrastructure/discovery/ServicePath.java (hunk 1 above)
Comment (dst) [inline micro-deps/micro-deps/src/main/java/com/ofg/infrastructure/discovery/ServicePath.java]: @cameleeck, how do you want to use here this method?

**Pull request closed without merging**